jac/jaclang/compiler/parser.py

```diff
@@ -1363,94 +1311,72 @@ def if_stmt(self, kid: list[ast.AstNode]) -> ast.IfStmt:
-                    ),
-                    kid=kid,
-                )
-            else:
-                raise self.ice()
+            self.consume_token(Tok.KW_ELIF)
+            condition = self.consume(ast.Expr)
+            body = self.consume(ast.SubNodeList)
+            else_body = self.match(ast.ElseStmt) or self.match(ast.ElseIf)
+            return ast.ElseIf(
+                condition=condition,
+                body=body,
+                else_body=else_body,
+                kid=kid,
+            )
 
         def else_stmt(self, kid: list[ast.AstNode]) -> ast.ElseStmt:
             """Grammar rule.
 
             else_stmt: KW_ELSE code_block
             """
-            if isinstance(kid[1], ast.SubNodeList):
-                return ast.ElseStmt(
-                    body=kid[1],
-                    kid=kid,
-                )
-            else:
-                raise self.ice()
+            self.consume_token(Tok.KW_ELSE)
+            body = self.consume(ast.SubNodeList)
+            return ast.ElseStmt(
+                body=body,
+                kid=kid,
+            )
 
         def try_stmt(self, kid: list[ast.AstNode]) -> ast.TryStmt:
             """Grammar rule.
 
             try_stmt: KW_TRY code_block except_list? else_stmt? finally_stmt?
             """
-            chomp = [*kid][1:]
-            block = chomp[0]
-            chomp = chomp[1:]
-            except_list = (
-                chomp[0]
-                if len(chomp) and isinstance(chomp[0], ast.SubNodeList)
-                else None
-            )
-            chomp = chomp[1:] if except_list else chomp
-            else_stmt = (
-                chomp[0] if len(chomp) and isinstance(chomp[0], ast.ElseStmt) else None
-            )
-            chomp = chomp[1:] if else_stmt else chomp
-            finally_stmt = (
-                chomp[0]
-                if len(chomp) and isinstance(chomp[0], ast.FinallyStmt)
-                else None
+            self.consume_token(Tok.KW_TRY)
+            block = self.consume(ast.SubNodeList)
+            except_list = self.match(ast.SubNodeList)
+            else_stmt = self.match(ast.ElseStmt)
+            finally_stmt = self.match(ast.FinallyStmt)
+            return ast.TryStmt(
+                body=block,
+                excepts=except_list,
+                else_body=else_stmt,
+                finally_body=finally_stmt,
+                kid=kid,
             )
-            if isinstance(block, ast.SubNodeList):
-                return ast.TryStmt(
-                    body=block,
-                    excepts=except_list,
-                    else_body=else_stmt,
-                    finally_body=finally_stmt,
-                    kid=kid,
-                )
-            else:
-                raise self.ice()
 
-        def except_list(self, kid: list[ast.AstNode]) -> ast.SubNodeList[ast.Except]:
+        def except_list(self, _: None) -> ast.SubNodeList[ast.Except]:
```

Comment:
This should be:

```python
items = [self.consume(ast.Except)]
while ex := self.match(ast.Except):
  items.append(ex)
return ast.Subnodelist[ast.Except]( items=items, ... )
```
1. No need to validate all the kids ast.Except, cause it's in the grammer and if not there will be error when consume or (future) when we leave the function we ensure all the nodes are processed thus any remaining nodes are not valid = error.
2. `self.ice()` shouldn't be called as it's now handled.